model/project_configs.go

```diff
@@ -36,7 +36,7 @@ type ProjectConfigFields struct {
 	TaskSync               *TaskSyncOptions               `yaml:"task_sync,omitempty" bson:"task_sync,omitempty"`
 	GithubTriggerAliases   []string                       `yaml:"github_trigger_aliases,omitempty" bson:"github_trigger_aliases,omitempty"`
 	PeriodicBuilds         []PeriodicBuildDefinition      `yaml:"periodic_builds,omitempty" bson:"periodic_builds,omitempty"`
-	ContainerSizes         map[string]ContainerResources  `yaml:"container_sizes,omitempty" bson:"container_sizes,omitempty"`
+	ContainerSizes         map[string]*ContainerResources `yaml:"container_sizes,omitempty" bson:"container_sizes,omitempty"`
```

Comment:
I wanted to be able to check in validation if a ContainerResources struct was empty without it being an error (since it should be ok for the resources property to be missing from a container if the size is defined in its place). Since memory and cpu at 0 are errors, I made this into a pointer to differentiate the two cases. But I think I can just change the resources field on Container to a pointer and not in this map.